src/newton_ac_powerflow.jl

```diff
@@ -0,0 +1,731 @@
+"""
+Solves a the power flow into the system and writes the solution into the relevant structs.
+Updates generators active and reactive power setpoints and branches active and reactive
+power flows (calculated in the From - To direction) (see
+[`flow_val`](@ref))
+
+Supports passing NLsolve kwargs in the args. By default shows the solver trace.
+
+Arguments available for `nlsolve`:
+
+  - `get_connectivity::Bool`: Checks if the network is connected. Default true
+  - `method` : See NLSolve.jl documentation for available solvers
+  - `xtol`: norm difference in `x` between two successive iterates under which
+    convergence is declared. Default: `0.0`.
+  - `ftol`: infinite norm of residuals under which convergence is declared.
+    Default: `1e-8`.
+  - `iterations`: maximum number of iterations. Default: `1_000`.
+  - `store_trace`: should a trace of the optimization algorithm's state be
+    stored? Default: `false`.
+  - `show_trace`: should a trace of the optimization algorithm's state be shown
+    on `STDOUT`? Default: `false`.
+  - `extended_trace`: should additifonal algorithm internals be added to the state
+    trace? Default: `false`.
+
+## Examples
+
+```julia
+solve_ac_powerflow!(sys)
+# Passing NLsolve arguments
+solve_ac_powerflow!(sys, method=:newton)
+```
+"""
+function solve_powerflow!(
+    pf::ACPowerFlow{<:ACPowerFlowSolverType},
+    system::PSY.System;
+    time_step::Int64 = 1,
+    kwargs...,
+)
+    #Save per-unit flag
+    settings_unit_cache = deepcopy(system.units_settings.unit_system)
```

Comment:
@rbolgaryn use `PSY.with_units_base` here and other few occurrences

Reply:
```julia

    with_units_base(exporter.system, PSY.UnitSystem.SYSTEM_BASE) do
        # Each of these corresponds to a group of records in the PSS/E spec
        for group_name in PSSE_GROUPS_33
            @debug "Writing export for group $group_name"
            write_to_buffers!(exporter, Val{Symbol(group_name)}())
        end
    end
```